src/index.js

```diff
@@ -147,15 +128,24 @@ function useCalendar({
 
   const date = new Date(referenceDate)
 
+  // Ensure events dates are stored as Date objects
+  events.map(event => {
+    const { date, startDate, endDate } = event;
+  
+    date && (event.date = new Date(event.date));
```

Comment:
Can you  either just use the de-structured variables within `new Date()` or else just use `event.<x>` everywhere?